jgrapht-core/src/main/java/org/jgrapht/alg/matching/KuhnMunkresMinimalWeightBipartitePerfectMatching.java

```diff
@@ -43,81 +43,85 @@
  * @author Alexey Kudinkin
  */
 public class KuhnMunkresMinimalWeightBipartitePerfectMatching<V, E>
-    implements WeightedMatchingAlgorithm<V, E>
+    implements MatchingAlgorithm<V, E>
 {
-    /////////////////////////////////////////////////////////////////////////////////////////////////
-
-    private final WeightedGraph<V, E> graph;
-
-    private final List<? extends V> firstPartition;
-    private final List<? extends V> secondPartition;
-
-    private final int[] matching;
+    private final Graph<V, E> graph;
+    private Set<? extends V> partition1;
+    private Set<? extends V> partition2;
 
     /**
      * Construct a new instance of the algorithm.
      * 
-     * @param G target weighted bipartite graph to find matching in
-     * @param S first vertex partition of the target bipartite graph
-     * @param T second vertex partition of the target bipartite graph
+     * @param graph the input graph
+     * @param partition1 the first partition of the vertex set
+     * @param partition2 the second partition of the vertex set
      */
     public KuhnMunkresMinimalWeightBipartitePerfectMatching(
-        final WeightedGraph<V, E> G, List<? extends V> S, List<? extends V> T)
+        Graph<V, E> graph, Set<? extends V> partition1, Set<? extends V> partition2)
+    {
+        if (graph == null) {
+            throw new IllegalArgumentException("Input graph cannot be null");
+        }
+        this.graph = graph;
+        if (partition1 == null) {
+            throw new IllegalArgumentException("Invalid partition provided");
```

Comment:
Could you change the message to: "Partition 1 cannot be null"
Same for partition 2.